backend/curation/api/curation-api.yml

```diff
@@ -560,6 +560,34 @@ paths:
                 items:
                   $ref: "#/components/schemas/dataset_index"
 
+  /v1/datasets/{dataset_id}/versions:
+    get:
+      summary: List all versions for a Dataset
+      tags:
+        - dataset
+      description: |
+        List all versions for a given Dataset. The `dataset_id` must reference a *Dataset*, **NOT** a *Dataset version*.
+        Dataset versions are returned in reverse-chronological order.
+      operationId: backend.curation.api.v1.curation.datasets.dataset_id.versions.actions.get
+      parameters:
+        - $ref: "#/components/parameters/path_dataset_id"
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  collection_id:
+                    $ref: "#/components/schemas/collection_id"
+                  datasets:
+                    type: array
+                    items:
+                      $ref: "#/components/schemas/dataset"
```

Comment:
This needs to be merged into a single object [for the sake of alphabetization](https://czi-sci.slack.com/archives/C02UGUV5R3J/p1678918244130559). It should be like [dataset_version](https://github.com/chanzuckerberg/single-cell-data-portal/blob/dan/rdev-4277/backend/curation/api/curation-api.yml#L979-L1067).

Reply:
If we're going to return `collection_id`, I really think we should also return `collection_version` idea, _especially_ for a `versions` endpoint.

Reply:
Its already alphabetized because I moved collection_id outside of the datasets array (its gonna be the same one for the whole list)

it'll be 
```
collection_id: ....
datasets: [
  ...,
  ...,
]
```

Reply:
re: collection_version_id--dataset versions can (and probably will) belong to more than 1 collection version. We can instead use that field to reference the collection version it was introduced in, but I would want to update the field name to be clear about that (and not accidentally imply it only exists in one collection version). Or, alternatively, provide a list of all the collection versions it does appear in. 

At that point, I do question how useful having it here is vs. providing the canonical collection ID and letting users get any follow up info by querying the collection version endpoint with the collection ID

Reply:
>Its already alphabetized because I moved collection_id outside of the datasets array (its gonna be the same one for the whole list)

Thanks, I missed this. Though this is making me think: we really should try to implement a consistent approach to the shape of the entities _across_ the endpoints. So for any endpoint that returns a list of Dataset-like objects, they should be the same, or as similar as possible.

So, if you're set on moving shared collection-level values like `collection_id` and `collection_version_id` up, out of the Dataset object, then we should probably do the same for the Datasets index endpoint: instead of an array of Datasets, return an array of Collection objects with Datasets arrange underneath, like with the Collections index endpoint (but with _full_ Dataset metadata instead). The question then is: do we really even need to maintain separate Collections and Datasets index endpoints? I think we might want to just consolidate into one. They're very similar, and returning the other Collection metadata on top of what we're already returning for the Datasets index endpoint is trivial. cc @brianraymor